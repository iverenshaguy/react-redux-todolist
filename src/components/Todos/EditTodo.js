import React from 'react';
import PropTypes from 'prop-types';
import Button from '../shared/Button';

const EditTodo = props => {
  return (
    <Button
      buttonType="button"
      btnClass="btn btn-outline-primary"
      buttonText="Edit Todo"
      onClick={props.onClick}
    />
  );
};

EditTodo.proptypes = {
  onClick: PropTypes.func.isRequired
};

export default EditTodo;
