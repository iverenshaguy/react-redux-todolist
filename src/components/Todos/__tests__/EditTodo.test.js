import EditTodo from '../EditTodo';

describe('EditTodo', () => {
  const clickFunction = jest.fn();

  it('renders correctly', () => {
    const wrapper = shallow(<EditTodo onClick={clickFunction} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('mounts correctly', () => {
    const wrapper = mount(<EditTodo onClick={clickFunction} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('calls a function on click', () => {
    const wrapper = shallow(<EditTodo onClick={clickFunction} />);

    wrapper.simulate('click');
    expect(clickFunction).toHaveBeenCalled();
  });
});
