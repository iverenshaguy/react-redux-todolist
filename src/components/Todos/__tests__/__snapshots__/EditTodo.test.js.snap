// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditTodo mounts correctly 1`] = `
<EditTodo
  onClick={[Function]}
>
  <Button
    btnClass="btn btn-outline-primary"
    buttonText="Edit Todo"
    buttonType="button"
    onClick={[Function]}
  >
    <button
      className="btn btn-outline-primary"
      onClick={[Function]}
      type="button"
    >
      Edit Todo
    </button>
  </Button>
</EditTodo>
`;

exports[`EditTodo renders correctly 1`] = `
<Button
  btnClass="btn btn-outline-primary"
  buttonText="Edit Todo"
  buttonType="button"
  onClick={[Function]}
/>
`;
